### ISSUE TITLE
Provide better exception message when no metadata file found

### ISSUE THREAD

**Issue by przepompownia** [schmittjoh/serializer#993]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no

## Steps required to reproduce the problem

1. Create metadata file for some class, check that (de)serialization works using it
2. Rename this file and try to (de)serialize.

## Expected Result
Avoiding using wrong type, and (more important) add related class name (and possibly expected metadata file paths when use Yaml or Xml driver) to exception message when no metadata found. 
* 

## Actual Result

*  Currently we get a message
```
TypeError : Argument 1 passed to JMS\Serializer\Context::pushClassMetadata() must be an instance of JMS\Serializer\Metadata\ClassMetadata, null given, called in /app/vendor/jms/serializer/src/JMS/Serializer/GraphNavigator.php on line 209
 /app/vendor/jms/serializer/src/JMS/Serializer/Context.php:222
 ...
```
that does not contain needed information. The simplest way is to create conditional (`null === $metadata`) breakpoint to get more information. 

It is also bug due to `TypeError` ocurrence.

**Comment by goetas** [comment]
The error you are describing should be related to something else.. but I agree that would be nice to have a better message in case of no meta data... the issue is that the last metadata driver is "annotation" that means that there are "always" metadata info for a class...

**Comment by przepompownia** [comment]
Indeed, I forgot to emphasize that in this example I have one-piece driver chain.

**Comment by goetas** [comment]
Why not appending at the end of the driver chain the nulldriver?

**Comment by przepompownia** [comment]
NullDriver at the end of chain does not resolve the problem of metadata missing by mistake. Maybe it make sense when we want to skip some metadata files. In this case element with no metadata is quietly skipped which is against the need for the information what metadata file not found.
